### PR TITLE
PHOENIX-6932 Update LAST_DDL_TIMESTAMP for index table when add/alter indexes

### PR DESCRIPTION
**Pull request by shahrs87**

### CODE REVIEW

**Comment by shahrs87** [comment]
Test `AlterTableIT#testSetPropertyDoesntUpdateDDLTimestamp` is failing after  PHOENIX-6930. Don't know how I missed observing the test failure. We don't need that test anymore so removing it.

**Comment by shahrs87** [comment]
@tkhurana  @gjacoby126  @virajjasani  Can you review the patch please? Thank you.

**Comment by shahrs87** [comment]
There is one crashing test:`org.apache.phoenix.util.IndexScrutinyIT` but when I run that locally it succeeds. 
For checkstyle errors, it is not related to the patch.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexIT.java [hunk 1]
```diff
@@ -1435,4 +1440,52 @@ public void testReturnedTimestamp() throws Exception {
...
+        String indexName = "IND_" + generateUniqueName();
+        String fullTableName = SchemaUtil.getTableName(TestUtil.DEFAULT_SCHEMA_NAME, tableName);
+        String fullIndexName = SchemaUtil.getTableName(TestUtil.DEFAULT_SCHEMA_NAME, indexName);
+
+        String ddl ="CREATE TABLE " + fullTableName + TestUtil.TEST_TABLE_SCHEMA + tableDDLOptions;
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.setAutoCommit(true);
+            Statement stmt = conn.createStatement();
+            stmt.execute(ddl);
+            ddl = "CREATE " + (localIndex ? "LOCAL" : "") + (uncovered ? "UNCOVERED" : "")
```
Comment (tkhurana) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexIT.java:1461]: Good to test the async index creation code path which first creates the index in building state and then call IndexTool to build the index which makes the index active. For example:

Comment (shahrs87) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexIT.java:1461]: @tkhurana  For example?

Comment (tkhurana) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexIT.java:1461]: @shahrs87 https://github.com/apache/phoenix/blob/master/phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java#L170-L174

Comment (shahrs87) [reply phoenix-core/src/it/java/org/apache/phoenix/end2end/index/BaseIndexIT.java:1461]: @tkhurana  Added a new test for ASYNC indexes. Can you please review again? Thank you.

**Review by tkhurana — APPROVED** [review]

**Comment by shahrs87** [comment]
There are 4 tests failing here 
https://ci-hadoop.apache.org/job/Phoenix/job/Phoenix-PreCommit-GitHub-PR/job/PR-1595/5/testReport/

```
org.apache.phoenix.end2end.index.UncoveredGlobalImmutableNonTxIndexIT.testLastDDLTimestampOnAsyncIndexes[UncoveredGlobalImmutableNonTxIndexIT_columnEncoded=false]
org.apache.phoenix.end2end.index.UncoveredGlobalImmutableNonTxIndexIT.testLastDDLTimestampOnAsyncIndexes[UncoveredGlobalImmutableNonTxIndexIT_columnEncoded=true]
org.apache.phoenix.end2end.index.UncoveredGlobalMutableNonTxIndexIT.testLastDDLTimestampOnAsyncIndexes[UncoveredGlobalMutableNonTxIndexIT_columnEncoded=false] org.apache.phoenix.end2end.index.UncoveredGlobalMutableNonTxIndexIT.testLastDDLTimestampOnAsyncIndexes[UncoveredGlobalMutableNonTxIndexIT_columnEncoded=true]
```
All are failing in Uncovered indexes. 
We create an ASYNC index and wait for it be in BUILDING state and then we run `IndexToolIT#runIndexTool` method to make it ACTIVE. But it looks like uncovered index doesn't go to BUILDING state.
Cc @tkhurana

**Comment by shahrs87** [comment]
Checkstyle errors are unrelated to the patch.
There is one test crashing on jenkins build:` org.apache.phoenix.end2end.join#SubqueryUsingSortMergeJoinIT` which runs successfully locally.
@tkhurana  Can you please merge this PR? Thank you.

**Pull request merged**